apps/aecore/lib/aecore/structures/signed_tx.ex

```diff
@@ -32,6 +32,38 @@ defmodule Aecore.Structures.SignedTx do
     key == nil && signature == nil
   end
 
+  @spec valid_byte_size?(SignedTx.t()) :: boolean()
+  def valid_byte_size?(%SignedTx{data: %{payload: %{oracle_address: address}}}) do
```

Comment:
I think there can be data_tx type matching and the specific validation happens inside the tx structure